File: gc-starter/crud-spring-boot-starter/src/main/java/com/gc/starter/crud/CrudAutoConfiguration.java
package com.gc.starter.crud;

import com.baomidou.mybatisplus.autoconfigure.MybatisPlusProperties;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;

import javax.annotation.PostConstruct;

/**
 * @author shizhongming
 * 2020/1/12 9:05 下午
 */
@Configuration
@ComponentScan
@Slf4j
@EnableConfigurationProperties(MybatisPlusProperties.class)
public class CrudAutoConfiguration {

    @PostConstruct
    public void init() {
        log.info("=========== crud 自动配置生效了 ===========");
    }

}

File: smart-starter/crud-spring-boot-starter/src/main/kotlin/com/smart/starter/crud/CrudAutoConfiguration.kt
package com.smart.starter.crud

import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import javax.annotation.PostConstruct

/**
 * 自动配置类
 * @author ming
 * 2019/9/14 下午6:22
 */
@Configuration
@ComponentScan
class CrudAutoConfiguration {

    @PostConstruct
    fun test () {
        println("=========== crud 自动配置生效了 ===========")
    }
}